WordPress/wordpress-develop
# Change wp_pattern_category public param to false

## What?
Sets the pattern categories taxonomy `public` param to `false` instead of true

## Why?
It was [suggested here](https://core.trac.wordpress.org/ticket/59569) that this would be a better setting for this taxonomy as it only needs to be accessed in admin, not the frontend.

Trac ticket: https://core.trac.wordpress.org/ticket/59569

## Testing Instructions
Log in as Admin, Editor, and Author users and make sure that each of these can add new and existing categories to patterns in the post editor and the site editor patterns library.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
